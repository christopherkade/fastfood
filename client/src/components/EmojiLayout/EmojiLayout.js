import React from "react"
import styled from "styled-components"

import { EmojiSelector } from "../EmojiSelector"
import data from "../../data/food.json"

const Wrapper = styled.ul`
  display: grid;
	grid-gap: 10px;
	grid-template-columns: repeat(auto-fit, minmax(80px, 1fr));
  grid-auto-flow: dense;
  margin: 0;
  padding: 0;
`

const EmojiLayout = ({ onClick }) => {
  return (
    <Wrapper>
      {data.foods.map(food => <EmojiSelector name={food.name} emoji={food.emoji} key={food.name} onClick={onClick} />)}
    </Wrapper>
  )
}

export default EmojiLayout
