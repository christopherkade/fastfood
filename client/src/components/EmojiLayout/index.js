export { default as EmojiLayout } from "./EmojiLayout"
