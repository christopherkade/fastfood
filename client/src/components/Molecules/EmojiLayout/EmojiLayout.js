import React from "react"
import PropTypes from 'prop-types';
import styled from "styled-components"

import { EmojiSelector } from "components/Atoms/EmojiSelector"
import data from "data/food.json"

const Wrapper = styled.ul`
  display: grid;
	grid-gap: 10px;
	grid-template-columns: repeat(auto-fit, minmax(80px, 1fr));
  grid-auto-flow: dense;
  margin: 0;
  padding: 0;
`

const EmojiLayout = ({ onClick, selections }) => {
  return (
    <Wrapper>
      {data.foods.map(food =>
        <EmojiSelector
          name={food.name}
          emoji={food.emoji}
          key={food.name}
          onClick={onClick}
          isSelected={selections.some(selection => selection.name === food.name)}
        />)}
    </Wrapper>
  )
}

EmojiLayout.propTypes = {
  onClick: PropTypes.func,
  selections: PropTypes.array
}

export default EmojiLayout
