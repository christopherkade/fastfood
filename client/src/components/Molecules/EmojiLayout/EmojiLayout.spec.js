import React from 'react'
import { render, cleanup } from 'react-testing-library'

import EmojiLayout from "./EmojiLayout"

afterEach(cleanup);

describe("<EmojiLayout />", () => {
  describe("Snapshots", () => {
    it("should match snapshot", () => {
      const { container } = render(<EmojiLayout onClick={() => { }} />);
      expect(container).toMatchSnapshot();
    });
  })
})
