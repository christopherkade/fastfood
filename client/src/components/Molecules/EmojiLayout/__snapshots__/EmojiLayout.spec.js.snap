// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmojiLayout /> Snapshots should match snapshot 1`] = `
<div>
  <ul
    class="sc-bwzfXH iPVXsn"
  >
    <button
      class="sc-bdVaJa evWdMy"
      title="Rice"
    >
      <span
        aria-label="Rice"
        role="img"
      >
        🍚
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Pizza"
    >
      <span
        aria-label="Pizza"
        role="img"
      >
        🍕
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Burger"
    >
      <span
        aria-label="Burger"
        role="img"
      >
        🍔
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Sushi"
    >
      <span
        aria-label="Sushi"
        role="img"
      >
        🍣
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Fries"
    >
      <span
        aria-label="Fries"
        role="img"
      >
        🍟
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Hotdog"
    >
      <span
        aria-label="Hotdog"
        role="img"
      >
        🌭
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Bacon"
    >
      <span
        aria-label="Bacon"
        role="img"
      >
        🥓
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Eggs"
    >
      <span
        aria-label="Eggs"
        role="img"
      >
        🥚
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Pancakes"
    >
      <span
        aria-label="Pancakes"
        role="img"
      >
        🥞
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Bread"
    >
      <span
        aria-label="Bread"
        role="img"
      >
        🍞
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Pretzel"
    >
      <span
        aria-label="Pretzel"
        role="img"
      >
        🥨
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Cheese"
    >
      <span
        aria-label="Cheese"
        role="img"
      >
        🧀
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Salad"
    >
      <span
        aria-label="Salad"
        role="img"
      >
        🥗
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Burito"
    >
      <span
        aria-label="Burito"
        role="img"
      >
        🥙
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Egg Sandwich"
    >
      <span
        aria-label="Egg Sandwich"
        role="img"
      >
        🥪
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Taco"
    >
      <span
        aria-label="Taco"
        role="img"
      >
        🌮
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Pork"
    >
      <span
        aria-label="Pork"
        role="img"
      >
        🍖
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Chicken"
    >
      <span
        aria-label="Chicken"
        role="img"
      >
        🍗
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Steak"
    >
      <span
        aria-label="Steak"
        role="img"
      >
        🥩
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Shrimp"
    >
      <span
        aria-label="Shrimp"
        role="img"
      >
        🦐
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Fish"
    >
      <span
        aria-label="Fish"
        role="img"
      >
        🐟
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Crab"
    >
      <span
        aria-label="Crab"
        role="img"
      >
        🦀
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Octopus"
    >
      <span
        aria-label="Octopus"
        role="img"
      >
        🐙
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Squid"
    >
      <span
        aria-label="Squid"
        role="img"
      >
        🦑
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Tomatoes"
    >
      <span
        aria-label="Tomatoes"
        role="img"
      >
        🍅
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Mushrooms"
    >
      <span
        aria-label="Mushrooms"
        role="img"
      >
        🍄
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Corn"
    >
      <span
        aria-label="Corn"
        role="img"
      >
        🌽
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Cucumber"
    >
      <span
        aria-label="Cucumber"
        role="img"
      >
        🥒
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Eggplant"
    >
      <span
        aria-label="Eggplant"
        role="img"
      >
        🍆
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Avocado"
    >
      <span
        aria-label="Avocado"
        role="img"
      >
        🥑
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Potatoes"
    >
      <span
        aria-label="Potatoes"
        role="img"
      >
        🥔
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Broccoli"
    >
      <span
        aria-label="Broccoli"
        role="img"
      >
        🥦
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Carrots"
    >
      <span
        aria-label="Carrots"
        role="img"
      >
        🥕
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Noodles"
    >
      <span
        aria-label="Noodles"
        role="img"
      >
        🥡
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Bento Box"
    >
      <span
        aria-label="Bento Box"
        role="img"
      >
        🍱
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Riceball"
    >
      <span
        aria-label="Riceball"
        role="img"
      >
        🍙
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Curry with Rice"
    >
      <span
        aria-label="Curry with Rice"
        role="img"
      >
        🍛
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Ramen"
    >
      <span
        aria-label="Ramen"
        role="img"
      >
        🍜
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Dumpling"
    >
      <span
        aria-label="Dumpling"
        role="img"
      >
        🥟
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Mushroom Soup"
    >
      <span
        aria-label="Mushroom Soup"
        role="img"
      >
        🍲
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Pasta"
    >
      <span
        aria-label="Pasta"
        role="img"
      >
        🍝
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Paella"
    >
      <span
        aria-label="Paella"
        role="img"
      >
        🥘
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Cereal"
    >
      <span
        aria-label="Cereal"
        role="img"
      >
        🥣
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Peanuts"
    >
      <span
        aria-label="Peanuts"
        role="img"
      >
        🥜
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Salted Popcorn"
    >
      <span
        aria-label="Salted Popcorn"
        role="img"
      >
        🍿
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Pie"
    >
      <span
        aria-label="Pie"
        role="img"
      >
        🥧
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Ice-cream"
    >
      <span
        aria-label="Ice-cream"
        role="img"
      >
        🍦
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Krispy Kreme"
    >
      <span
        aria-label="Krispy Kreme"
        role="img"
      >
        🍩
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Chips Ahoy"
    >
      <span
        aria-label="Chips Ahoy"
        role="img"
      >
        🍪
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Chocolate Cake"
    >
      <span
        aria-label="Chocolate Cake"
        role="img"
      >
        🎂
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Strawberry Shortcake"
    >
      <span
        aria-label="Strawberry Shortcake"
        role="img"
      >
        🍰
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Chocolate"
    >
      <span
        aria-label="Chocolate"
        role="img"
      >
        🍫
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Sweeties"
    >
      <span
        aria-label="Sweeties"
        role="img"
      >
        🍬
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Pudding"
    >
      <span
        aria-label="Pudding"
        role="img"
      >
        🍮
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Kiwi"
    >
      <span
        aria-label="Kiwi"
        role="img"
      >
        🥝
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Grapes"
    >
      <span
        aria-label="Grapes"
        role="img"
      >
        🍇
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Watermelon"
    >
      <span
        aria-label="Watermelon"
        role="img"
      >
        🍉
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Oranges"
    >
      <span
        aria-label="Oranges"
        role="img"
      >
        🍊
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Bananas"
    >
      <span
        aria-label="Bananas"
        role="img"
      >
        🍌
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Pineapples"
    >
      <span
        aria-label="Pineapples"
        role="img"
      >
        🍍
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Apples"
    >
      <span
        aria-label="Apples"
        role="img"
      >
        🍎
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Strawberries"
    >
      <span
        aria-label="Strawberries"
        role="img"
      >
        🍓
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Peaches"
    >
      <span
        aria-label="Peaches"
        role="img"
      >
        🍑
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Pears"
    >
      <span
        aria-label="Pears"
        role="img"
      >
        🍐
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Cherries"
    >
      <span
        aria-label="Cherries"
        role="img"
      >
        🍒
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Milk"
    >
      <span
        aria-label="Milk"
        role="img"
      >
        🥛
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Coffee"
    >
      <span
        aria-label="Coffee"
        role="img"
      >
        ☕
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Tea"
    >
      <span
        aria-label="Tea"
        role="img"
      >
        🍵
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Coke"
    >
      <span
        aria-label="Coke"
        role="img"
      >
        🥤
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Beer"
    >
      <span
        aria-label="Beer"
        role="img"
      >
        🍺
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Wine"
    >
      <span
        aria-label="Wine"
        role="img"
      >
        🍷
      </span>
    </button>
    <button
      class="sc-bdVaJa evWdMy"
      title="Apple Cider"
    >
      <span
        aria-label="Apple Cider"
        role="img"
      >
        🍾
      </span>
    </button>
  </ul>
</div>
`;
